actors/evm/src/interpreter/precompiles.rs

```diff
@@ -69,23 +111,39 @@ const fn gen_precompiles<RT: Runtime>() -> [PrecompileFn<RT>; 13] {
         lookup_address,     // resolve_address 0x0b
         get_actor_code_cid, // get code cid 0x0c
         get_randomness,     // rand 0x0d
+        call_actor,         // call_actor 0x0e
     ]
 }
 
 pub struct Precompiles<RT>(PhantomData<RT>);
 
 impl<RT: Runtime> Precompiles<RT> {
-    const PRECOMPILES: [PrecompileFn<RT>; 13] = gen_precompiles();
+    const PRECOMPILES: [PrecompileFn<RT>; 14] = gen_precompiles();
     const MAX_PRECOMPILE: U256 = {
         let mut limbs = [0u64; 4];
         limbs[0] = Self::PRECOMPILES.len() as u64;
         U256(limbs)
     };
 
-    pub fn call_precompile(runtime: &RT, precompile_addr: U256, input: &[u8]) -> PrecompileResult {
-        Self::PRECOMPILES[precompile_addr.0[0] as usize - 1](runtime, input)
+    // Precompile Context will be flattened to None if not calling the call_actor precompile
+    pub fn call_precompile(
+        runtime: &RT,
+        precompile_addr: U256,
+        input: &[u8],
+        context: PrecompileContext,
+    ) -> PrecompileResult {
+        let context =
+            if Self::is_call_actor_precompile(&precompile_addr) { Some(context) } else { None };
+
+        Self::PRECOMPILES[precompile_addr.0[0] as usize - 1](runtime, input, context)
+    }
+
+    #[inline]
+    pub fn is_call_actor_precompile(addr: &U256) -> bool {
+        addr.0[0] == 0x0e
     }
```

Comment:
We can just always pass the context, right? This seems error prone and confusing.

Reply:
removed